pkg/placement/controllers/scheduling/scheduling_controller.go

```diff
@@ -430,141 +435,187 @@ func newMisconfiguredCondition(status *framework.Status) metav1.Condition {
+// bind will also return the decision groups for placement status.
 func (c *schedulingController) bind(
 	ctx context.Context,
 	placement *clusterapiv1beta1.Placement,
-	clusterDecisions []clusterapiv1beta1.ClusterDecision,
+	decisionGroups ScheduleDecisionGroups,
 	clusterScores PrioritizerScore,
 	status *framework.Status,
-) error {
-	// sort clusterdecisions by cluster name
-	sort.SliceStable(clusterDecisions, func(i, j int) bool {
-		return clusterDecisions[i].ClusterName < clusterDecisions[j].ClusterName
-	})
-
-	// split the cluster decisions into slices, the size of each slice cannot exceed
-	// maxNumOfClusterDecisions.
-	decisionSlices := [][]clusterapiv1beta1.ClusterDecision{}
-	remainingDecisions := clusterDecisions
-	for index := 0; len(remainingDecisions) > 0; index++ {
-		var decisionSlice []clusterapiv1beta1.ClusterDecision
-		switch {
-		case len(remainingDecisions) > maxNumOfClusterDecisions:
-			decisionSlice = remainingDecisions[0:maxNumOfClusterDecisions]
-			remainingDecisions = remainingDecisions[maxNumOfClusterDecisions:]
-		default:
-			decisionSlice = remainingDecisions
-			remainingDecisions = nil
-		}
-		decisionSlices = append(decisionSlices, decisionSlice)
-	}
-	// if decisionSlices is empty, append one empty slice.
-	// so that can create a PlacementDecision with empty decisions in status.
-	if len(decisionSlices) == 0 {
-		decisionSlices = append(decisionSlices, []clusterapiv1beta1.ClusterDecision{})
-	}
-
-	// bind cluster decision slices to placementdecisions.
+) ([]clusterapiv1beta1.DecisionGroupStatus, error) {
 	errs := []error{}
-
+	placementDecisionIndex := 0
 	placementDecisionNames := sets.NewString()
-	for index, decisionSlice := range decisionSlices {
-		placementDecisionName := fmt.Sprintf("%s-decision-%d", placement.Name, index+1)
-		placementDecisionNames.Insert(placementDecisionName)
-		err := c.createOrUpdatePlacementDecision(
-			ctx, placement, placementDecisionName, decisionSlice, clusterScores, status)
-		if err != nil {
-			errs = append(errs, err)
+	decisionGroupStatus := []clusterapiv1beta1.DecisionGroupStatus{}
+
+	// generate at least on empty decisionGroup, this is to ensure there's an empty placement decision if no cluster selected.
+	if len(decisionGroups) == 0 {
+		decisionGroups = append(decisionGroups, ScheduleDecisionGroup{})
+	}
+
+	// create/update placement decision for each decision group
+	for decisionGroupIndex, decisionGroup := range decisionGroups {
+		// sort clusterdecisions by cluster name
+		clusterDecisions := decisionGroup.ClusterDecisions
+		sort.SliceStable(clusterDecisions, func(i, j int) bool {
+			return clusterDecisions[i].ClusterName < clusterDecisions[j].ClusterName
+		})
+
+		// generate placement decisions, group index starts from 0, decision name index starts from 1 for readability.
+		pds, groupStatus := c.generatePlacementDecisionsAndStatus(placement, decisionGroup, decisionGroupIndex, placementDecisionIndex+1)
```

Comment:
this is weird. the two should at least start with the same index

Reply:
change to start from 0.